wpical/src/main/native/cpp/tagpose.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <tagpose.h>
+
+namespace tag {
+pose::pose(double xpos, double ypos, double zpos, double w, double x, double y,
+           double z) {
+  xPos = xpos;
+  yPos = ypos;
+  zPos = zpos;
+  quaternion = Eigen::Quaterniond(w, x, y, z);
+  rotationMatrix = quaternion.toRotationMatrix();
+  transformMatrixFmap.setZero();
+  transformMatrixFmap.block<3, 3>(0, 0) = rotationMatrix;
+  transformMatrixFmap(0, 3) = xpos - (16.541 / 2.0);
+  transformMatrixFmap(1, 3) = ypos - (8.211 / 2.0);
```

Comment:
These look like magic constants--what do they represent?  Should they actually be constant, or passed in from somewhere else?